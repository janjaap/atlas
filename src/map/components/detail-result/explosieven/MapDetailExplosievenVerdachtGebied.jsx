import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailExplosievenVerdachtGebied = ({
  panoUrl, verdachtGebied, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={verdachtGebied.label}
    title="Verdacht gebied"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Hoofdgroep"
        value={verdachtGebied.type}
      />
      <MapDetailResultItem
        label="Subsoort"
        value={verdachtGebied.subType}
      />
      <MapDetailResultItem
        label="Opmerkingen"
        value={verdachtGebied.remarks}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailExplosievenVerdachtGebied.propTypes = {
  verdachtGebied: PropTypes.shape({
    label: PropTypes.string,
    remarks: PropTypes.string,
    subType: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailExplosievenVerdachtGebied;
