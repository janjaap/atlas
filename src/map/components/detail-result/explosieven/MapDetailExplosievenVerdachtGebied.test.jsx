import React from 'react';
import { shallow } from 'enzyme';

import MapDetailExplosievenVerdachtGebied from './MapDetailExplosievenVerdachtGebied';

describe('MapDetailExplosievenVerdachtGebied', () => {
  it('should render everything', () => {
    const verdachtGebied = {
      label: 'Verdacht gebied label',
      remarks: 'Verdacht gebied remarks',
      subType: 'Verdacht gebied sub type',
      type: 'Verdacht gebied type'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailExplosievenVerdachtGebied
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        verdachtGebied={verdachtGebied}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
