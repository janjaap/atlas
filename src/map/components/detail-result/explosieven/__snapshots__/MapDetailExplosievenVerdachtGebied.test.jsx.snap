// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailExplosievenVerdachtGebied should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Verdacht gebied label"
  title="Verdacht gebied"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Hoofdgroep"
      value="Verdacht gebied type"
    />
    <MapDetailResultItem
      label="Subsoort"
      value="Verdacht gebied sub type"
    />
    <MapDetailResultItem
      label="Opmerkingen"
      value="Verdacht gebied remarks"
    />
  </ul>
</MapDetailResultWrapper>
`;
